LLama/Native/LLamaTokenDataArray.cs

```diff
@@ -342,7 +362,7 @@ public struct LLamaTokenDataArrayNative
         /// A pointer to an array of LlamaTokenData
         /// </summary>
         /// <remarks>Memory must be pinned in place for all the time this LLamaTokenDataArrayNative is in use</remarks>
-        public IntPtr data;
+        public unsafe LLamaTokenData* data;
```

Comment:
Is it possible to avoid exposing this member? It's a raw pointer so it's a bit dangerous to expose it to users.

Reply:
This is the _very_ low level structure that just directly represents the equivalent llama.cpp struct so it has to be like this (I actually consider an explicit pointer _safer_ than an IntPtr, since at least it carries type data with it. wheras an IntPtr is a `void*` in disguise).

Reply:
Right that makes sense. What about adding a comment to tell users not to modify the data? Though `LLamaTokenData*` is more type-safe but I'm afraid that users may write code like `tokenDataArray.data[0] = new LLamaTokenData(...);`. That seems to be bit shocking! :D

Reply:
It's pretty hard to get your hands on a `LLamaTokenDataArrayNative` in the first place - the higher level `LLamaTokenDataArray` is what's normally exposed through APIs. You would have to deliberately construct one and then the only way you could use it would be to directly call the low level APIs.

Reply:
@AsakusaRinne I realised I can do even better than a comment and changed the whole thing over to a span, so it's completely safe!